### ISSUE TITLE
Crater runs for beta 1.31

### ISSUE THREAD

**Issue by pietroalbini** [rust-lang/rust#55540]
cc @rust-lang/release

**Comment by pietroalbini** [comment]
@craterbot run name=beta-1.31-1 start=1.30.0 end=beta-2018-10-30 cap-lints=warn p=1

**Comment by Mark-Simulacrum** [comment]
Just as a note this beta is odd since it bootstraps from an older stable than it should I believe

**Comment by cuviper** [comment]
The current beta inherited nightly's bootstrap from 1.30-beta (2018-10-13), rather than using 1.30 stable, but that's not much difference.  #55550 updates this.

**Comment by Mark-Simulacrum** [comment]
Yes, it shouldn't have any impact on the crater run but wanted to make that note.

**Comment by Mark-Simulacrum** [comment]
@craterbot run name=beta-1.31-2 start=1.30.0 end=beta-2018-11-13 cap-lints=warn p=5

**Comment by pietroalbini** [comment]
@craterbot start=1.30.1

**Comment by pietroalbini** [comment]
@craterbot run start=1.30.1 end=beta-2018-11-17 mode=rustdoc cap-lints=warn p=5

**Comment by pietroalbini** [comment]
@craterbot abort name=pr-55540

**Comment by pietroalbini** [comment]
@craterbot run name=beta-1.31-3 start=1.30.1 end=beta-2018-11-17 mode=rustdoc cap-lints=warn p=5

**Comment by pietroalbini** [comment]
Triaged the rustdoc run (`beta-1.31-3`), the other run (`beta-1.31-2`) still needs triage AFAIK.

**Comment by alexcrichton** [comment]
Analysis of [beta-1.31-2](https://github.com/rust-lang/rust/issues/55540#issuecomment-439641400):

* [assert-cli](https://crater-reports.s3.amazonaws.com/beta-1.31-2/beta-2018-11-13/reg/assert_cli-0.6.3/log.txt) - couldn't reproduce
* [bmp085](https://crater-reports.s3.amazonaws.com/beta-1.31-2/beta-2018-11-13/reg/bmp085-0.1.2/log.txt) - couldn't reproduce
* [cage](https://crater-reports.s3.amazonaws.com/beta-1.31-2/beta-2018-11-13/reg/cage-0.2.6/log.txt) - https://github.com/rust-lang/rust/issues/56082
* [completable_future](https://crater-reports.s3.amazonaws.com/beta-1.31-2/beta-2018-11-13/reg/completable_future-0.1.0/log.txt) - couldn't reproduce
* [crux](https://crater-reports.s3.amazonaws.com/beta-1.31-2/beta-2018-11-13/reg/crux-0.3.0/log.txt) - couldn't reproduce
* [ecdh](https://crater-reports.s3.amazonaws.com/beta-1.31-2/beta-2018-11-13/reg/ecdh-0.0.12/log.txt) - couldn't reproduce
* [environment](https://crater-reports.s3.amazonaws.com/beta-1.31-2/beta-2018-11-13/reg/environment-0.1.1/log.txt) - couldn't reproduce
* [etclient](https://crater-reports.s3.amazonaws.com/beta-1.31-2/beta-2018-11-13/reg/etclient-0.3.0/log.txt) - not a root regression
* [evm-stateful](https://crater-reports.s3.amazonaws.com/beta-1.31-2/beta-2018-11-13/reg/evm-stateful-0.9.2/log.txt) - https://github.com/rust-lang/rust/issues/56083
* [expanduser](https://crater-reports.s3.amazonaws.com/beta-1.31-2/beta-2018-11-13/reg/expanduser-1.2.1/log.txt) - couldn't reproduce
* [hey_listen](https://crater-reports.s3.amazonaws.com/beta-1.31-2/beta-2018-11-13/reg/hey_listen-0.2.0/log.txt) - couldn't reproduce
* [message_filter](https://crater-reports.s3.amazonaws.com/beta-1.31-2/beta-2018-11-13/reg/message_filter-0.6.0/log.txt) - couldn't reproduce
* [pidlock](https://crater-reports.s3.amazonaws.com/beta-1.31-2/beta-2018-11-13/reg/pidlock-0.1.2/log.txt) - couldn't reproduce
* [postgis](https://crater-reports.s3.amazonaws.com/beta-1.31-2/beta-2018-11-13/reg/postgis-0.6.0/log.txt) - https://github.com/rust-lang/rust/issues/56084
* [rustc_llvm_proxy](https://crater-reports.s3.amazonaws.com/beta-1.31-2/beta-2018-11-13/reg/rustc-llvm-proxy-0.1.4/log.txt) - couldn't reproduce
* [rusty_dashed](https://crater-reports.s3.amazonaws.com/beta-1.31-2/beta-2018-11-13/reg/rusty_dashed-0.2.1/log.txt) - https://github.com/rust-lang/rust/issues/56082
* [silk](https://crater-reports.s3.amazonaws.com/beta-1.31-2/beta-2018-11-13/reg/silk-0.3.3/log.txt) - couldn't reproduce
* [sputnikvm-stateful](https://crater-reports.s3.amazonaws.com/beta-1.31-2/beta-2018-11-13/reg/sputnikvm-stateful-0.10.0/log.txt) - https://github.com/rust-lang/rust/issues/56083
* [strip-ansi-escapes](https://crater-reports.s3.amazonaws.com/beta-1.31-2/beta-2018-11-13/reg/strip-ansi-escapes-0.1.0/log.txt) - couldn't reproduce
* [structopt-flags](https://crater-reports.s3.amazonaws.com/beta-1.31-2/beta-2018-11-13/reg/structopt-flags-0.1.0/log.txt) - couldn't reproduce
* [type_level_examples](https://crater-reports.s3.amazonaws.com/beta-1.31-2/beta-2018-11-13/reg/type_level_examples-0.0.22/log.txt) - https://github.com/rust-lang/rust/issues/56087
* [vectortile](https://crater-reports.s3.amazonaws.com/beta-1.31-2/beta-2018-11-13/reg/vectortile-0.2.2/log.txt) - not a root regression
* [wombo](https://crater-reports.s3.amazonaws.com/beta-1.31-2/beta-2018-11-13/reg/wombo-0.1.2/log.txt) - couldn't reproduce
* [sputnikvm](https://crater-reports.s3.amazonaws.com/beta-1.31-2/beta-2018-11-13/gh/ETCDEVTeam.sputnikvm-dev/log.txt) - not a root regression
* [points-api-importer](https://crater-reports.s3.amazonaws.com/beta-1.31-2/beta-2018-11-13/gh/OSM-Browser.points-api-importer/log.txt) - not a root regression
* [cita-crate-genesis](https://crater-reports.s3.amazonaws.com/beta-1.31-2/beta-2018-11-13/gh/TheWaWaR.cita-create-genesis/log.txt) - not a root regression
* [cage](https://crater-reports.s3.amazonaws.com/beta-1.31-2/beta-2018-11-13/gh/faradayio.cage/log.txt) - already reported
* [etclient](https://crater-reports.s3.amazonaws.com/beta-1.31-2/beta-2018-11-13/gh/sorpaas.etclient/log.txt) - not a root regression
* [tokio-by-hand](https://crater-reports.s3.amazonaws.com/beta-1.31-2/beta-2018-11-13/gh/casey.tokio-by-hand/log.txt) - couldn't reproduce
* [plato](https://crater-reports.s3.amazonaws.com/beta-1.31-2/beta-2018-11-13/gh/fromheten.plato/log.txt) - couldn't reproduce
* [padd](https://crater-reports.s3.amazonaws.com/beta-1.31-2/beta-2018-11-13/gh/srhickma.padd/log.txt) - couldn't reproduce
* [rip_media](https://crater-reports.s3.amazonaws.com/beta-1.31-2/beta-2018-11-13/gh/ssokolow.rip_media/log.txt) - couldn't reproduce

**Comment by Aaron1011** [comment]
All of the rustdoc regressions from [beta-1.31-3](https://crater-reports.s3.amazonaws.com/beta-1.31-3/index.html) appear to actually be build/check errors, rather than actual `rustdoc` issues.

**Comment by Aaron1011** [comment]
It appears that some crates spuriously errored in the `rustdoc` run. For example, the [futures-signals crate](https://github.com/Pauan/rust-signals) (the Github repository is called `rust-signals`) errored with this message:

```
Nov 18 21:51:03.555 ERRO this task or one of its parent failed!
Nov 18 21:51:03.555 ERRO command `"/mnt/big/crater/work/local/cargo-home/bin/cargo" "+1.30.1" "fetch" "--locked" "--manifest-path" "Cargo.toml"` failed
```

However, documenting this crate locally produces the error described in https://github.com/rust-lang/rust/issues/50159

**Comment by pietroalbini** [comment]
@craterbot run nane=beta-1.31-4 start=1.30.1 end=beta-2018-11-24 cap-lints=warn p=5

**Comment by pietroalbini** [comment]
@craterbot run name=beta-1.31-4 start=1.30.1 end=beta-2018-11-24 cap-lints=warn p=5

**Comment by alexcrichton** [comment]
I've found two regression in that last report: https://github.com/rust-lang/rust/issues/56316 and https://github.com/rust-lang/rust/issues/56317

**Comment by pietroalbini** [comment]
Last beta run for this cycle!

@craterbot run start=1.30.1 end=master#8940af3cea4a9a9deabcaf843b2f8cad9a83fc61 mode=build-and-test cap-lints=warn p=5

**Comment by pietroalbini** [comment]
@craterbot abort

**Comment by pietroalbini** [comment]
@craterbot run name=beta-1.31-5 start=1.30.1 end=master#8940af3cea4a9a9deabcaf843b2f8cad9a83fc61 mode=build-and-test cap-lints=warn p=5

**Comment by pietroalbini** [comment]
No new regression found (thanks @Centril and @alexcrichton for helping triaging the results!).

**Comment by nikic** [comment]
Scary LLVM assertions:

crushtool:

> Dec 03 14:51:48.263 INFO [stderr] rustc: /checkout/src/llvm/lib/IR/Type.cpp:241: static llvm::IntegerType *llvm::IntegerType::get(llvm::LLVMContext &, unsigned int): Assertion `NumBits >= MIN_INT_BITS && "bitwidth too small"' failed.

simdnoise:

> Dec 02 16:48:59.627 INFO [stderr] rustc: /checkout/src/llvm/lib/Target/X86/MCTargetDesc/X86MCCodeEmitter.cpp:118: static uint8_t (anonymous namespace)::X86MCCodeEmitter::ModRMByte(unsigned int, unsigned int, unsigned int): Assertion `Mod < 4 && RegOpcode < 8 && RM < 8 && "ModRM Fields out of range!"' failed.

**Comment by alexcrichton** [comment]
@nikic agreed! Some investigation figured out those weren't regressions though and date back as far as 1.28.0. (we don't have cached builds further back than that for rustc builds linking against an LLVM with assertions enabled) Those two assertions were spawned off into https://github.com/rust-lang/rust/issues/56482 and https://github.com/rust-lang/rust/issues/56483

**Comment by pietroalbini** [comment]
A note for why this is happening: this Crater run did use a start toolchain assertions disabled (1.30.1) and an end toolchain with assertions enabled (the beta commit), so all the llvm assertions caught were marked as regressions.

**Comment by alexcrichton** [comment]
@nikic the easiest way is likely using [this crate](https://crates.io/crates/rustup-toolchain-install-master) with the `--alt` CLI flag
